jade/errors.py

```diff
@@ -1,4 +1,144 @@
+import sys
+import traceback
 
 
-class TrustedClientVerificationError(RuntimeError):
-    pass
+class RequestError(RuntimeError):
+    "An error occured inside of JADE while processing a request"
+    TYPE = None
```

Comment:
Hmm.  It seems that "None" is the right value here.  Missing will cause a crash (AttributeError).  We could check to see if the attribute exists before trying to access the value.  Ultimately this will need to be represent-able in JSON.  None is convenient because None --> null.  If we were to use something like NotImplemented, then it would error when converting to JSON.

It seems to me that **intentionally missing** == **not specified** and preparing for **lost somehow** seems overbearing.  Can you imagine a scenario where the TYPE or SUBTYPE is **lost somehow**?

Reply:
I see now that you were proposing strings containing "unspecified_type" and "unspecified_subtype".  It seems that one could use "not_specified" when they'd like to intentionally not specify whereas intentionally missing would be None/null.

Reply:
If you feel strongly about this, no problem.  My perspective is just that error handling needs to be absolutely hailstormproof, and leaving the potential for a value to be either string or null is risky for both the server and client.  You're right about "unspecified" sending the wrong signal, perhaps "unimplemented_type" would be more clear, if we decide to take this string-only approach?

Reply:
The whole point of `null`/`None` is mixing it with other types.  Are you against the use of `null`/`None` in general?

Reply:
We can pick this up later, no I'm not opposed to `null` in general, but in this specific case I am opposed.  I'd like the error to be clear and not up for misinterpretation.